test/onnx/verify/add_fp8_test.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/float8.hpp>
+#include <migraphx/register_target.hpp>
+#include <migraphx/verify.hpp>
+#include <onnx_test.hpp>
+
+TEST_CASE(add_fp8_test)
+{
+    auto p = optimize_onnx("add_fp8_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape s{migraphx::shape::fp8e4m3fnuz_type, {1}};
+
+    migraphx::parameter_map pp;
+    migraphx::literal l1{s, {3.25}};
+    migraphx::literal l2{s, {2.25}};
+    pp["0"] = l1.get_argument();
+    pp["1"] = l2.get_argument();
+
+    auto result = p.eval(pp).back();
+    std::vector<migraphx::fp8::fp8e4m3fnuz> result_vector;
+    result.visit([&](auto output) { result_vector.assign(output.begin(), output.end()); });
+    std::vector<migraphx::fp8::fp8e4m3fnuz> gold{static_cast<migraphx::fp8::fp8e4m3fnuz>(5.5)};
```

Comment:
Does it really require `static_cast` ?

Reply:
Without this, I get the following error:
```
/code/AMDMIGraphX/AMDMIGraphX/test/onnx/verify/add_fp8_test.cpp:46:50: error: type 'double' cannot be narrowed to 'size_type' (aka 'unsigned long') in initializer list [-Wc++11-narrowing]
    std::vector<migraphx::fp8::fp8e4m3fnuz> gold{5.5};
                                                 ^~~
/code/AMDMIGraphX/AMDMIGraphX/test/onnx/verify/add_fp8_test.cpp:46:50: note: insert an explicit cast to silence this issue
    std::vector<migraphx::fp8::fp8e4m3fnuz> gold{5.5};
                                                 ^~~
                                                 static_cast<size_type>( )
1 error generated.
```